diff -ru ptw-base/_ptw32.h ptw-gen/_ptw32.h
--- a/_ptw32.h	2026-08-27 06:02:38.428787160 +0000
+++ b/_ptw32.h	2026-08-27 06:02:38.464787576 +0000
@@ -56,7 +56,7 @@
 
 #if defined(__GNUC__)
 # pragma GCC system_header
-# if ! defined __declspec
+# if ! defined __declspec && ! defined __clang__
 #  error "Please upgrade your GNU compiler to one that supports __declspec."
 # endif
 #endif
diff -ru ptw-base/create.c ptw-gen/create.c
--- a/create.c	2026-08-27 06:02:38.428787160 +0000
+++ b/create.c	2026-08-27 06:02:38.460787530 +0000
@@ -226,7 +226,7 @@
    * finished with it here.
    */
 
-#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__)
+#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__) || defined(WINCE)
 
   tp->threadH =
       threadH =
diff -ru ptw-base/implement.h ptw-gen/implement.h
--- a/implement.h	2026-08-27 06:02:38.428787160 +0000
+++ b/implement.h	2026-08-27 06:02:38.464787576 +0000
@@ -740,7 +740,7 @@
 
   void PTW32_CDECL ptw32_rwlock_cancelwrwait (void *arg); /* matches type ptw32_cleanup_callback_t this way */
 
-#if ! defined (__MINGW32__) || (defined (__MSVCRT__) && ! defined (__DMC__))
+#if ! defined (__MINGW32__) || (defined (__MSVCRT__) && ! defined (__DMC__)) || defined(WINCE)
   unsigned __stdcall
 #else
   void
@@ -859,7 +859,8 @@
 # define PTW32_TO_VLONG64PTR(ptr) (ptr)
 #endif
 
-#if defined(__GNUC__)
+#if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__) || \
+    defined(_M_IX86) || defined(_M_X64))
 # if defined(_WIN64)
 # define PTW32_INTERLOCKED_COMPARE_EXCHANGE_64(location, value, comparand) \
     ({                                                                     \
diff -ru ptw-base/pthread_exit.c ptw-gen/pthread_exit.c
--- a/pthread_exit.c	2026-08-27 06:02:38.428787160 +0000
+++ b/pthread_exit.c	2026-08-27 06:02:38.460787530 +0000
@@ -95,7 +95,7 @@
        * Implicit POSIX handles are cleaned up in ptw32_throw() now.
        */
 
-#if ! defined (__MINGW32__) || defined (__MSVCRT__)  || defined (__DMC__)
+#if ! defined (__MINGW32__) || defined (__MSVCRT__)  || defined (__DMC__) || defined(WINCE)
       _endthreadex ((unsigned) (size_t) value_ptr);
 #else
       _endthread ();
diff -ru ptw-base/ptw32_threadDestroy.c ptw-gen/ptw32_threadDestroy.c
--- a/ptw32_threadDestroy.c	2026-08-27 06:02:38.428787160 +0000
+++ b/ptw32_threadDestroy.c	2026-08-27 06:02:38.460787530 +0000
@@ -56,7 +56,7 @@
       /*
        * Copy thread state so that the thread can be atomically NULLed.
        */
-#if ! defined(__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__)
+#if ! defined(__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__) || defined(WINCE)
       HANDLE threadH = tp->threadH;
 #endif
       HANDLE cancelEvent = tp->cancelEvent;
@@ -74,7 +74,7 @@
 	  CloseHandle (cancelEvent);
 	}
 
-#if ! defined(__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__)
+#if ! defined(__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__) || defined(WINCE)
       /*
        * See documentation for endthread vs endthreadex.
        */
diff -ru ptw-base/ptw32_threadStart.c ptw-gen/ptw32_threadStart.c
--- a/ptw32_threadStart.c	2026-08-27 06:02:38.428787160 +0000
+++ b/ptw32_threadStart.c	2026-08-27 06:02:38.464787576 +0000
@@ -118,7 +118,7 @@
 # pragma optimize("g", off)
 #endif
 
-#if ! defined (__MINGW32__) || (defined (__MSVCRT__) && ! defined (__DMC__))
+#if ! defined (__MINGW32__) || (defined (__MSVCRT__) && ! defined (__DMC__)) || defined(WINCE)
 unsigned
   __stdcall
 #else
@@ -151,7 +151,7 @@
 
   free (threadParms);
 
-#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__)
+#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__) || defined(WINCE)
 #else
   /*
    * _beginthread does not return the thread id and is running
@@ -308,7 +308,7 @@
   (void) pthread_win32_thread_detach_np ();
 #endif
 
-#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__)
+#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__) || defined(WINCE)
   _endthreadex ((unsigned)(size_t) status);
 #else
   _endthread ();
@@ -318,7 +318,7 @@
    * Never reached.
    */
 
-#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__)
+#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__) || defined(WINCE)
   return (unsigned)(size_t) status;
 #endif
 
diff -ru ptw-base/ptw32_throw.c ptw-gen/ptw32_throw.c
--- a/ptw32_throw.c	2026-08-27 06:02:38.428787160 +0000
+++ b/ptw32_throw.c	2026-08-27 06:02:38.464787576 +0000
@@ -97,7 +97,7 @@
        * explicit thread exit here after cleaning up POSIX
        * residue (i.e. cleanup handlers, POSIX thread handle etc).
        */
-#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__)
+#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__) || defined(WINCE)
       unsigned int exitCode = 0;
 
       switch (exception)
@@ -120,7 +120,7 @@
 
 #endif
 
-#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__)
+#if ! defined (__MINGW32__) || defined (__MSVCRT__) || defined (__DMC__) || defined(WINCE)
       _endthreadex (exitCode);
 #else
       _endthread ();
